tls/ciphersuites.h

```diff
@@ -115,101 +80,19 @@ typedef struct {
 	int				max_major_ver;
 	int				max_minor_ver;
 	unsigned char			flags;
+	TlsMpiPool			*mpi_profile[__TTLS_ECP_DP_N];
```

Comment:
Although type says "pool", variable name is about "profile". That may be interpreted as equivalence of the two terms. But as I understood from the code, profiles are not really about memory, but about the blueprints of the data, if I can put it that way.
It would be nice to have a little more clarity here. Unfortunately, I can't offer an acceptable solution.

Reply:
Profiles are pools actually: we have the cipher suites static MPI pools and use then on the module initialization - allocate all necessary PK contexts. In run time, durinag a handshake, we create per-handshake MPI pool and just copy an appropriate cipher suite profile. Since we copy using SIMD, this is much faster than initialize each MPI, even with the allocation using the pools.

I wrote small comment about this at the header of `mpool.c`:
> The pool also used for MPI profiles (see below) to initialize a profile
> implicitly for MPI math. Dynamically allocated pages are used instead of
> static per-cpu ones.